File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/MedicinalProductIndication.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class MedicinalProductIndication : DomainResource, IResource
{
    public Base.CodeableConcept? DiseaseSymptomProcedure { get; set; }
    public Base.ResourceReference[]? UndesirableEffect { get; set; }
    public Base.Quantity? Duration { get; set; }
    public MedicinalProductIndication_OtherTherapy[]? OtherTherapy { get; set; }
    public Base.CodeableConcept[]? Comorbidity { get; set; }
    public Base.CodeableConcept? IntendedEffect { get; set; }
    public Base.Population[]? Population { get; set; }
    public Base.CodeableConcept? DiseaseStatus { get; set; }
    public Base.ResourceReference[]? Subject { get; set; }

    public class MedicinalProductIndication_OtherTherapy : BackboneElement
    {
        public object? Medication    
        {
            get
            {
                if (MedicationReference is not null)
                {
                    return MedicationReference;
                }
        
                if (MedicationCodeableConcept is not null)
                {
                    return MedicationCodeableConcept;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    MedicationReference = (Base.ResourceReference)value;
                    return;
                }
        
                if (value?.GetType() == typeof(Base.CodeableConcept))
                {
                    MedicationCodeableConcept = (Base.CodeableConcept)value;
                    return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.ResourceReference? MedicationReference { get; set; }
        public required Base.CodeableConcept TherapyRelationshipType { get; set; }
        public Base.CodeableConcept? MedicationCodeableConcept { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/MedicinalProductIndication.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import Population
from base import Quantity
from base import DomainResource
from base import Reference
from base import BackboneElement

@dataclass(kw_only=True)
class MedicinalProductIndicationOtherTherapy(BackboneElement):
    medication_reference: Optional[Reference] = None
    therapy_relationship_type: CodeableConcept
    medication_codeable_concept: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class MedicinalProductIndication(DomainResource):
    disease_symptom_procedure: Optional[CodeableConcept] = None
    undesirable_effect: Optional[List[Reference]] = field(default_factory=list)
    duration: Optional[Quantity] = None
    other_therapy: Optional[List[MedicinalProductIndicationOtherTherapy]] = field(default_factory=list)
    comorbidity: Optional[List[CodeableConcept]] = field(default_factory=list)
    intended_effect: Optional[CodeableConcept] = None
    population: Optional[List[Population]] = field(default_factory=list)
    disease_status: Optional[CodeableConcept] = None
    subject: Optional[List[Reference]] = field(default_factory=list)